src/rd.h

```diff
@@ -446,3 +447,5 @@ typedef struct rd_chariov_s {
         char  *ptr;
         size_t size;
 } rd_chariov_t;
+
+#endif /* NDef RD_H */
```

Comment:
iow I'd be more comfortable with #endif /* <filename>_H */
presumably the format <filename>_H carry an implicite 'this is the end of the header guard'
and implicit notation is better than incorrect one :-)